ml-agents/mlagents/trainers/trainer/rl_trainer.py

```diff
@@ -80,6 +81,41 @@ def _is_ready_update(self):
         """
         return False
 
+    def _policy_mean_reward(self) -> Optional[float]:
+        """ Returns the mean episode reward for the current policy. """
+        rewards = self.cumulative_returns_since_policy_update
+        if len(rewards) == 0:
+            return None
+        else:
+            return sum(rewards) / len(rewards)
+
+    @timed
+    def _checkpoint(self) -> None:
+        """
+        Saves the model
+        """
+        n_policies = len(self.policies.keys())
+        if n_policies > 1:
+            logger.warning(
+                "Trainer has multiple policies, but default behavior only saves the first."
+            )
+        policy = list(self.policies.values())[0]
+        policy.checkpoint(self._policy_mean_reward())
+
+    def save_model(self) -> None:
+        """
+        Exports the model
+        """
+        n_policies = len(self.policies.keys())
+        if n_policies > 1:
+            logger.warning(
+                "Trainer has multiple policies, but default behavior only saves the first."
```

Comment:
This code is only here to make this work for the ghost trainer I believe (since it is the only one (for now) that needs a Dict[str, Policy]), but I think it should be the responsibility of the ghost trainer to know what policy it needs to save rather than saving "the first one".

Reply:
Related to the comment above about `Trainer.policies`.  There is no `Trainer.policy` but I was hoping to avoid following the rabbit hole of adding that and deprecating `Trainer.policies` in this PR.